pkg/local_object_storage/engine/delete.go

```diff
@@ -121,7 +121,11 @@ func (e *StorageEngine) delete(prm DeletePrm) (DeleteRes, error) {
 
 func (e *StorageEngine) deleteChildren(addr oid.Address, force bool, splitID *objectSDK.SplitID) {
 	var fs objectSDK.SearchFilters
-	fs.AddSplitIDFilter(objectSDK.MatchStringEqual, splitID)
+	if splitID != nil {
```

Comment:
Is it legitimate to to call `deleteChildren()` with `nil` `splitID`?

Reply:
i left previous behavior untouched. Overall, in current implementation it's safe cuz objects w/o parent ("small") don't appear in `CID_splitid` bucket so they are unprocessed. But it'd be more safe to not call `deleteChildren` at all in this case

Reply:
I'd make it `splitID objectSDK.SplitID` as well and let the upper layer handle it (not a lot of these layers here).

Reply:
done